src/components/ha-wa-dialog.ts

```diff
@@ -0,0 +1,380 @@
+
+  @property({ type: Boolean, reflect: true })
+  public open = false;
+
+  @property({ type: String, reflect: true, attribute: "width" })
+  public width: DialogWidth = "medium";
+
+  @property({
+    type: String,
+    reflect: true,
+    attribute: "width-on-title-click",
+  })
+  public widthOnTitleClick: DialogWidthOnTitleClick = "none";
+
+  @property({ type: Boolean, reflect: true, attribute: "prevent-scrim-close" })
+  public preventScrimClose = false;
+
+  @property({ type: String, attribute: "header-title" })
+  public headerTitle = "";
+
+  @property({ type: String, attribute: "header-subtitle" })
+  public headerSubtitle = "";
+
+  @property({ type: String, attribute: "header-subtitle-position" })
+  public headerSubtitlePosition: "above" | "below" = "below";
+
+  @property({ type: Boolean, reflect: true, attribute: "flexcontent" })
+  public flexContent = false;
+
+  @state()
+  private _open = false;
+
+  @state()
+  private _sizeChanged = false;
+
+  protected updated(
+    changedProperties: Map<string | number | symbol, unknown>
+  ): void {
+    super.updated(changedProperties);
+
+    if (changedProperties.has("open")) {
+      this._open = this.open;
+    }
+
+    if (changedProperties.has("width")) {
+      this._sizeChanged = false;
+    }
+
+    this.classList.toggle("size-changed", this._sizeChanged);
+  }
+
+  protected render() {
+    return html`
+      <wa-dialog
+        .open=${this._open}
+        .lightDismiss=${!this.preventScrimClose}
+        without-header
+        @wa-show=${this._handleShow}
+        @wa-after-hide=${this._handleAfterHide}
+      >
+        <slot name="heading">
+          <ha-dialog-header .subtitlePosition=${this.headerSubtitlePosition}>
+            <slot name="navigationIcon" slot="navigationIcon">
+              <ha-icon-button
+                data-dialog="close"
+                .label=${this.hass?.localize("ui.common.close") ?? "Close"}
+                .path=${mdiClose}
+              ></ha-icon-button>
+            </slot>
+            ${this.headerTitle
+              ? html`<span
+                  slot="title"
+                  class="title"
+                  @click=${this.toggleWidth}
+                >
+                  ${this.headerTitle}
+                </span>`
+              : nothing}
+            ${this.headerSubtitle
+              ? html`<span slot="subtitle">${this.headerSubtitle}</span>`
+              : nothing}
+            <slot name="actionItems" slot="actionItems"></slot>
+          </ha-dialog-header>
+        </slot>
+        <div class="body ha-scrollbar">
+          <slot></slot>
+        </div>
+        <slot name="footer" slot="footer"></slot>
+      </wa-dialog>
+    `;
+  }
+
+  private _handleShow = () => {
+    this._open = true;
+    fireEvent(this, "opened");
+
+    this.updateComplete.then(() => {
+      const focusElement = this.querySelector(
+        "[dialogInitialFocus]"
+      ) as HTMLElement;
+      focusElement?.focus();
+    });
+  };
```

Comment:
```suggestion
  private async _handleShow = () => {
    this._open = true;
    fireEvent(this, "opened");

    await this.updateComplete
    const focusElement = this.querySelector(
      "[dialogInitialFocus]"
    ) as HTMLElement;
    focusElement?.focus();
  };
```

so we don't use native `autofocus` as webawesome is doing it?